paasta_tools/contrib/get_containers_and_ips.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+import argparse
+import os
+import socket
+
+from paasta_tools.utils import get_docker_client
+
+HAPROXY_STATS_SOCKET = '/var/run/synapse/haproxy.sock'
+
+
+def send_to_haproxy(command):
+    s = socket.socket(socket.AF_UNIX)
+    # 1 seconds should be more than enough of a timeout since HAProxy is local
+    s.settimeout(1)
+    s.connect(HAPROXY_STATS_SOCKET)
+    print("Sending {} to HAProxy stats socket".format(command))
+    s.send((command + '\n').encode())
+    file_handle = s.makefile()
+    response = file_handle.read().splitlines()
+    print("Response: {}".format(response))
+    s.close()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=(
+            'Script to dump a HAProxy map between container IPs and task IDs.'
+        ),
+    )
+    parser.add_argument(
+        'map_file', nargs='?', default='/var/run/synapse/maps/ip_to_service.map',
+        help='Where to write the output map file',
+    )
+    args = parser.parse_args()
+
+    client = get_docker_client()
+
+    file_exists = os.path.isfile(args.map_file)
+    if file_exists:
+        mode = 'r+'
+    else:
+        mode = 'w'
+
+    with open(args.map_file, mode) as f:
```

Comment:
This is way too much for a main function to do. Instead make a function to read in the contents of the file in this format and make it do the opening and have it return the data, returning and empty {} if the file doesn't exist.